pkg/hub/controllers/placemanifestworkcontroller/placemanifestwork_deploy_reconcile.go

```diff
@@ -18,5 +27,92 @@ type deployReconciler struct {
 
 func (d *deployReconciler) reconcile(ctx context.Context, pw *workapiv1alpha1.PlaceManifestWork) (*workapiv1alpha1.PlaceManifestWork, reconcileState, error) {
 	// TODO the main manifestwork create/update/delete logic, we may also need to manage the status condition for whether manifestwork is created here
+	klog.Info("reconciling placeManifestWork ", pw.Name)
+
+	placement, err := d.placementLister.Placements(pw.Namespace).Get(pw.Spec.PlacementRef.Name)
+	if err != nil {
+		//klog.Info("my reconciling placement err ", err)
+		apimeta.SetStatusCondition(&pw.Status.Conditions, GetPlacementDecisionVerified(workapiv1alpha1.PlacementDecisionNotFound, ""))
+
+		return pw, reconcileContinue, fmt.Errorf("Failed get placement %w", err)
+	}
+
+	//klog.Info("my reconciling placement ", placement)
+	// TODO: Could be better to change the pw status to include list of selected clusters otherwise we need to list all the manifestworks belong to pw to figure which cluster Added or deleted
+	req, _ := labels.NewRequirement(PlaceManifestWorkControllerNameLabelKey, selection.In, []string{pw.Name})
+	selector := labels.NewSelector().Add(*req)
+
+	manifestWorks, _ := d.manifestWorkLister.List(selector)
+	var existingClusters sets.String
+	for _, mw := range manifestWorks {
+		existingClusters.Insert(mw.Namespace)
+	}
+
+	added, deleted, _ := helper.GetClusters(d.placeDecisionLister, placement, existingClusters)
+	//klog.Info("Selected cluters count ", len(added), len(deleted))
+
+	if len(added) == 0 && len(existingClusters) == 0 {
+		apimeta.SetStatusCondition(&pw.Status.Conditions, GetPlacementDecisionVerified(workapiv1alpha1.PlacementDecisionEmpty, ""))
+		apimeta.SetStatusCondition(&pw.Status.Conditions, GetManifestworkApplied(workapiv1alpha1.PlacementDecisionEmpty, ""))
+	}
+
+	if len(deleted) > 0 {
+		// TODO Delete manifestwork
+	}
+
+	for cluster := range added {
+		mw, err := CreateManifestWork(pw, cluster)
```

Comment:
this should be applyManifestWork. We would need to treat manifestwork update. you can use this https://github.com/open-cluster-management-io/api/blob/main/utils/work/v1/workapplier/workapplier.go

we can init the the applier during init the reconcile.